gamemodes/sss/core/itemtype/hats.pwn

```diff
@@ -111,6 +111,13 @@ stock SetPlayerHatItem(playerid, Item:itemid)
 
 	RemoveItemFromWorld(itemid);
 	RemoveCurrentItem(GetItemHolder(itemid));
+	
+	if(hat_CurrentHatItem[playerid] == itemid)
+		return 1;
+	    
+   	if(IsValidItem(hat_CurrentHatItem[playerid]))
+    		GiveWorldItemToPlayer(playerid, hat_CurrentHatItem[playerid]);
```

Comment:
I think this wants a return after because if this path is taken, the hat will be returned to the player but `hat_CurrentHatItem` will still be assigned